src/satellite.jl

```diff
@@ -1,48 +1,54 @@
+module Satellite
 using gps
-using LinearAlgebra
-using gps.Satellite
-const pi = gps.pi
-const π = gps.pi
+const π = gps.π
 
-"""
-    satloc(sat, ℓ, t)
-Find the cartesian coordinates and time of the satellite such that its
-broadcast at that location will reach coordinates `ℓ` at time `t`.
-"""
-function satloc(sat::Sat, xv::Coordinates, tv::Real)::Tuple{Coordinates,<:Real}
-    # This function should have a root at the time we desire
-    function f(t::Real)::Real
-        Δx = Satellite.position(sat,t)-xv
-        Δx⋅Δx-c^2*(tv-t)^2
+struct Sat
+    u::Coordinates
+    v::Coordinates
+    periodicity::Real
+    altitude::Real
+    phase::Real
+    Sat(u,v,p,a,θ) = begin
+        validatecoords(u,v)
+        new(u,v,p,a,θ)
+    end
+end
+# add some nice tersity to access
+function Base.getproperty(x::Sat, s::Symbol)
+    if s===:θ
+        getfield(x, :phase)
+    elseif s===:h
+        getfield(x, :altitude)
+    elseif s===:p
+        getfield(x, :periodicity)
+    else 
+        getfield(x, s)
     end
-    # We start our search at the current vehicle time, assuming it will be rather close
-    # Furthermore our δ must be less than .01m/c in order to have 1cm accuracy
-    t = Newton.newton(f, tv, δ=BigFloat(.01)/c)
-    Satellite.position(sat, t), t
 end
 
+open(DATA_FILE) do f
+    lines = map(s -> :(BigFloat($(split(s)[1]))), readlines(f))
+    sats=[]
+    for i in 5:9:length(lines)
+        ex = :(Sat([$(lines[i:i+2]...)],
+                   [$(lines[i+3:i+5]...)],
+                   $(lines[i+6]),
+                   $(lines[i+7]),
+                   $(lines[i+8])))
+        push!(sats, ex)
+    end
+    eval(:(const satellites = [$(sats...)]))
+end
+
+export Sat, satellites, position
+
```

Comment:
A much cleaner way of doing this, have you tested?

Reply:
yep! all tests pass
might be good to add in CI so this can be verified